### DISCUSSION TITLE
Exported Code

### DISCUSSION THREAD

**Discussion by LalFX** [Akascape/CTkDesigner-Support#1] [Q&A]
Can you share the actual code that is exported from the app?

**Comment by Akascape** [comment]
@LalFX 
```python

import tkinter
import customtkinter
from PIL import Image

customtkinter.set_appearance_mode("System") 
customtkinter.set_default_color_theme("saved_theme.json")
customtkinter.set_widget_scaling(1.0)

HEIGHT = 400
WIDTH = 500
                
app = customtkinter.CTk()
app.title("My App")
app.geometry((f"{WIDTH}x{HEIGHT}"))
app.resizable(False, False)


Button1 = customtkinter.CTkButton(master=app, width=246, height=44, corner_radius=22, text='SIGN IN')
Button1.place(x=133, y=241)

Entry2 = customtkinter.CTkEntry(master=app, width=340, height=48, justify='center', placeholder_text='USER ID')
Entry2.place(x=81, y=93)

Label3 = customtkinter.CTkLabel(master=app, font=customtkinter.CTkFont('Roboto',30), text='LOGIN PAGE')
Label3.place(x=163, y=31)

Entry4 = customtkinter.CTkEntry(master=app, width=340, height=48, justify='center', placeholder_text='PASS')
Entry4.place(x=83, y=165)

CheckBox5 = customtkinter.CTkCheckBox(master=app, text='Remember me')
CheckBox5.place(x=201, y=303)

app.mainloop()
                
```
Example I created in [this video](https://youtu.be/bIWLkiYeWFg)

**Reply by Akascape** [reply]
@LalFX CTkDesigner doesn't add default parameters in widget, only the changed parameters. + It can also store ctkfont/ctkimage data.

**Reply by LalFX** [reply]
Thanks for sharing. Looks like an interesting addon for a pure WYSIWYG experience. Congrats!

Can you auto fill elements to the width of the frame? Also is there any other platform other than kofi to try the app?

**Reply by Akascape** [reply]
@LalFX 

> Can you auto fill elements to the width of the frame? Also is there any other platform other than kofi to try the app?

Auto fill is not currently implemented, but duplicating widget is there. And maybe I will add autofill in the next version.

Currently I have uploaded it on kofi only with a membership access which also includes lots of other useful code templates and projects. https://ko-fi.com/akascape/shop

**Reply by SoothingMist** [reply]
In tkinter, there is a means of stating which subroutine to execute when a button is pushed. For instance: ResetNode.configure(command=GUI_support.On_ResetNode)

What is the equivalent in CustomTkinter? Is it necessary to state that manually for each button? It should be one of the settable parameters at the GUI specification level.

**Reply by Akascape** [reply]
@SoothingMist You can pass the parameter as "command=..." in the manual parameter entry box, but this approach is not recommended since CTkDesigner cannot directly implement function code. Instead, it’s better to add the command and function manually in your exported code, for example:
`
ResetNode.configure(command=GUI_support.On_ResetNode)
`
That said, I plan to introduce a code addition feature in the application in the next version.

**Reply by SoothingMist** [reply]
Ok. Good. Thanks. Currently using PAGE (http://page.sourceforge.net) but it does not support graphs and other components that CTkDesigner seems to support. Will give CTkDesigner a shot during the next iteration of my project (https://github.com/SoothingMist/Scalable-Point-to-Point-LoRa-Sensor-Network). Hoping to write an eBook on this exploration and will mention CTkDesigner in Next-Steps.
